dali/operators/decoder/video/video_decoder_cpu.cc

```diff
@@ -12,66 +12,143 @@
+FFmpeg is used for decoding. For Mixed backend, NVIDIA's Video Codec SDK (NVDEC) is used.
+Each output sample is a sequence of frames with shape ``(F, H, W, C)`` where:
+
+* ``F`` is the number of frames in the sequence (can vary between samples)
+* ``H`` is the frame height in pixels
+* ``W`` is the frame width in pixels
+* ``C`` is the number of color channels
+
+Example 1: Extract a sequence of arbitrary frames:
+
+.. code-block:: python
+
+    video_decoder = dali.experimental.decoders.video(
+        encoded=encoded_video,
+        frames=[0, 10, 20, 30, 40, 50, 40, 30, 20, 10, 0]
+        ...,
+    )
+
+Example 2: Extract a sequence of evenly spaced frames, starting from frame 0,
+with a stride of 2, until 10 frames are reached:
+
+.. code-block:: python
+
+    video_decoder = dali.experimental.decoders.Video(
+        encoded=encoded_video,
+        start_frame=0, sequence_length=10, stride=2
+        ...,
+    )
+
+Example 3: Pad the sequence with the last frame in the video, until 100 frames are reached:
+
+.. code-block:: python
+
+    video_decoder = dali.experimental.decoders.Video(
+        encoded=encoded_video,
+        start_frame=0, sequence_length=100, stride=2, pad_mode="edge"
+        ...,
+    )
+
+Example 4: Pad the sequence with a constant value of 128, until 100 frames are reached:
 
-The video streams can be in most of the container file formats. FFmpeg is used to parse video
- containers and returns a batch of sequences of frames with shape (F, H, W, C) where F is the
- number of frames in a sequence and can differ for each sample.)code")
+.. code-block:: python
+
+    video_decoder = dali.experimental.decoders.Video(
+        encoded=encoded_video,
+        start_frame=0, sequence_length=100, stride=2, pad_mode="constant", fill_value=128
+        ...,
+
+Example 5: Pad the sequence with a constant RGB value of (118, 185, 0), until 100 frames are reached:
+
+.. code-block:: python
+
+    video_decoder = dali.experimental.decoders.Video(
+        encoded=encoded_video,
+        start_frame=0, sequence_length=100, stride=2, pad_mode="constant", fill_value=[118, 185, 0]
+        ...,
+)code")
     .NumInput(1)
     .NumOutput(1)
-    .InputDox(0, "buffer", "TensorList", "Data buffer with a loaded video file.")
+    .InputDox(0, "encoded", "TensorList", "Encoded video stream")
     .AddOptionalArg("affine",
-    R"code(Applies only to the mixed backend type.
+                    R"code(Whether to pin threads to CPU cores (mixed backend only).
+
+If True, each thread in the internal thread pool will be pinned to a specific CPU core.
+If False, threads can migrate between cores based on OS scheduling.)code",
+                    true)
+    .AddOptionalArg<std::vector<int>>(
+        "frames",
+        R"code(Specifies which frames to extract from each video by their indices.
+
+The indices can be provided in any order and can include duplicates. For example, ``[0,10,5,10]`` would extract:
```

Comment:
If pad is not supported I would mention that all frames needs to be inside the video.

Reply:
It is supported now